docs/proposals/002-api-proposal/proposal.md

```diff
@@ -80,101 +80,99 @@ The API design is based on these axioms:
 // condition, one will be selected at random. 
-type LLMService struct {
+type InferenceModel struct {
         metav1.ObjectMeta
         metav1.TypeMeta
 
-        Spec LLMServiceSpec
+        Spec InferenceModelSpec
 }
 
-type LLMServiceSpec struct {
-        // Defines the distinct services.
-        // Model can be in 2 priority classes, Critical and Noncritical. 
-        // Priority class is implicitly set to Critical by specifying an Objective.
-        // Otherwise the Model is considered Noncritical.
-        Models []Model
-        // Reference to the backend pools that the services registers to.
-        PoolRef []corev1.ObjectReference
-}
-
-// Model defines the policies for routing the traffic of a use case, this includes performance objectives 
-// and traffic splitting between different versions of the model.
-type Model struct {
+type InferenceModelSpec struct {
         // The name of the model as the users set in the "model" parameter in the requests.
-        // The name should be unique among the services that reference the same backend pool.
+        // The name should be unique among the workloads that reference the same backend pool.
         // This is the parameter that will be used to match the request with. In the future, we may
         // allow to match on other request parameters. The other approach to support matching on 
         // on other request parameters is to use a different ModelName per HTTPFilter.
         // Names can be reserved without implementing an actual model in the pool.
         // This can be done by specifying a target model and setting the weight to zero,
         // an error will be returned specifying that no valid target model is found.
-        Name string
+        ModelName string
         // Optional
-        // LLM Services with an objective have higher priority than services without.
-        // IMPORTANT: By specifying an objective, this places the LLMService in a higher priority class than LLMServices without a defined priority class. 
-        // In the face of resource-scarcity. Higher priority requests will be preserved, and lower priority class requests will be rejected.
-        Objective *Objective
+        // Defines how important it is to serve the model compared to other models referencing the same pool.
+        Criticality *Criticality
         // Optional.
-	      // Allow multiple versions of a model for traffic splitting. 
-	      // If not specified, the target model name is defaulted to the modelName parameter.
+	    // Allow multiple versions of a model for traffic splitting. 
+	    // If not specified, the target model name is defaulted to the modelName parameter.
         // modelName is often in reference to a LoRA adapter.
         TargetModels []TargetModel
+        // Reference to the InferencePool that the model registers to. It must exist in the same namespace.
+        PoolRef InferencePoolReference
```

Comment:
Sorry if I missed the discussion. We previously had a list of PoolRefs, is it intentional changing it to one?